bin/node/src/commands.rs

```diff
@@ -0,0 +1,168 @@
+use crate::chain_spec::{self, get_account_id_from_seed, AuthorityKeys, LOCAL_AUTHORITIES};
+use crate::cli::{Cli, ExtraParams};
+use aleph_primitives::AuthorityId as AlephId;
+use log::info;
+use sc_cli::{CliConfiguration, Error, KeystoreParams, SharedParams};
+use sc_keystore::LocalKeystore;
+use sc_service::config::{BasePath, KeystoreConfig};
+use sp_application_crypto::key_types;
+use sp_consensus_aura::sr25519::AuthorityId as AuraId;
+use sp_core::crypto::KeyTypeId;
+use sp_core::{ed25519, sr25519};
+use sp_keystore::SyncCryptoStore;
+use std::io::Write;
+use std::sync::Arc;
+use structopt::StructOpt;
+
+/// The `bootstrap-chain` command is used generate keys for the genesis authorities
+/// keys are written to the keystore of the authorities
+/// and the chain specification is printed to stdout in the JSON format
+#[derive(Debug, StructOpt)]
+pub struct BootstrapChainCmd {
+    /// Force raw genesis storage output.
+    #[structopt(long = "raw")]
+    pub raw: bool,
+
+    #[structopt(flatten)]
+    pub keystore_params: KeystoreParams,
+
+    #[structopt(flatten)]
+    pub shared_params: SharedParams,
+}
+
+impl CliConfiguration for BootstrapChainCmd {
+    fn shared_params(&self) -> &SharedParams {
+        &self.shared_params
+    }
+}
+
+impl BootstrapChainCmd {
+    fn upsert_aura_key(
+        &self,
+        keystore: &LocalKeystore,
+        key_type: KeyTypeId,
+    ) -> Result<AuraId, Error> {
```

Comment:
There are several things with this method.
0. I don't like the `upsert` name as we never update an existing value. Can we rename it to something like `get_aura_key`?
1. We don't use `self` at all, so can make it a "static" method or even a top-level function,
2. We don't really make use of the exact type of `keystore`, so I think it would be cleaner to use the type  `impl SyncCryptoStore`,
3. We only call it with `key_type = key_types::AURA`, and it wouldn't make sense to call it with any other name given the value, so I think we could get rid of it.
4.  the return type is `Result`, but we never return `Err(_)` variant, so we can simply return `AuraId`.

These comments apply to `upsert_aleph_key` too.

Reply:
Ad 0) Intended behaviour of this code is to insert a new key into the keystore or retrieve an existing one - get_or_set?
Ad 1) Done
Ad 2) I don't follow this comment - open-keystore returns an Arc<LocalKeystore> which is a type, not a trait?
Ad 3) Done
Ad 4) Done